spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/orm/jpa/AbstractJpaAutoConfigurationTests.java

```diff
@@ -77,10 +80,40 @@ public void close() {
 	public void testNoDataSource() throws Exception {
 		this.context.register(PropertyPlaceholderAutoConfiguration.class,
 				getAutoConfigureClass());
-		this.expected.expect(BeanCreationException.class);
-		this.expected.expectMessage("No qualifying bean");
-		this.expected.expectMessage("DataSource");
 		this.context.refresh();
+		assertThat(this.context.getBeansOfType(JpaProperties.class).size()).isEqualTo(1);
```

Comment:
Hi @snicoll , i have a question for your opinion.

Should i change to `assertThat(this.context.getBeansOfType(JpaProperties.class)).isEmpty();` ? or other ?